cluster/cluster.go

```diff
@@ -23,7 +24,7 @@ type Cluster struct {
 	localAddr string
 }
 
-func Join(ctx context.Context, cfg Config) (*Cluster, error) {
+func Join(ctx context.Context, cfg *Config, clientUrls []string) (*Cluster, error) {
```

Comment:
Is there a reason why `clientURLs` was added in the method signature instead of the config struct? I think the latter would be cleaner

Reply:
we need client urls from the initial cluster nodes, they aren't present in the config struct of new nodes being added

Reply:
can we add this to our cluster config then?